### ISSUE TITLE
Linking redis container doesn't survive containers restart

### ISSUE THREAD

**Issue by rutsky** [tutumcloud/hipache#2]
`run.sh` uses environment variables for obtaining linked Redis container address and port, which is not updated if container is being restarted [1].

This most probably leads to malfunction of hipache after docker daemon restart, if Redis is used from separate container.

Proper way to address linked Redis container is using it's hostname, generated from link alias (as described [2]): "redis" hostname (linked container alias) and 6379 port (AFAIK port is the same as in Redis container).

[1] https://docs.docker.com/userguide/dockerlinks/#important-notes-on-docker-environment-variables
[2] https://docs.docker.com/userguide/dockerlinks/#updating-the-etc-hosts-file

**Comment by fermayo** [comment]
True. Will just need to replace `REDIS_HOST=$REDIS_PORT_6379_TCP_ADDR` with `REDIS_HOST=redis` on `run.sh`. Can you give it a go and test it?

**Comment by rutsky** [comment]
Sure. What about `elif [ ! -z "$REDIS_1_PORT_6379_TCP_ADDR" ]; then` branch?
Is it used or I can remove it?

**Comment by fermayo** [comment]
That's for compatibility with old docker-compose/fig binaries - I think we can remove it

**Comment by rutsky** [comment]
Oh, looks like currently Hipache 0.3.1 is broken. When I build my own image and start Hipache I get:

```
util.js:555
  ctor.prototype = Object.create(superCtor.prototype, {
                                          ^
TypeError: Cannot read property 'prototype' of undefined
    at Object.exports.inherits (util.js:555:43)
    at Object.<anonymous> (/usr/local/lib/node_modules/hipache/node_modules/http-proxy/lib/http-proxy/index.js:111:17)
    at Module._compile (module.js:456:26)
    at Object.Module._extensions..js (module.js:474:10)
    at Module.load (module.js:356:32)
    at Function.Module._load (module.js:312:12)
    at Module.require (module.js:364:17)
    at require (module.js:380:17)
    at Object.<anonymous> (/usr/local/lib/node_modules/hipache/node_modules/http-proxy/lib/http-proxy.js:4:17)
    at Module._compile (module.js:456:26)

```

Looks like this is following bug https://github.com/hipache/hipache/issues/225
Hipache 1.0.3 depends on fixed version of `http-proxy` (1.0.2) which depends on any version of `eventemitter3` (`*`) which latest version is not compatible with `http-proxy` (1.0.2).
So technically this is `http-proxy` bug.

What do you suggest to fix this?

I saw suggestions to either upgrade `http-proxy` to version, that have fixed `eventemitter3` dependency, or to force `eventemitter3` version.
